tests/test_fields.py

```diff
@@ -1349,6 +1350,92 @@ def test_numeric_field_astype(self):
             self.assertTrue(isinstance(context.exception,TypeError))
 
 
+class TestFieldUnique(unittest.TestCase):
```

Comment:
Can we have in order and out of order for first appearance of values relative to their sorted order. Might be worth doing both for each specialized type.